added export of images to assets

steeleb opened the pull request:
Woof! Apparently exporting image collections to assets is just downright terrible. 

__Create a validation image collection__
1) creation of the 4-7 validation workflow. `eePlumB/Z_PrepAOI/validationSet_L457.js`
2) edits to the validation dataset for 4-7 and 8-9 to reflect new AOI tiles. 
3) absolute hack job of creating the image collection for validation. I'm not even sure if this is going to work yet, I've had images processing for about an hour now, so cross your fingers. If this doesn't work, I'll bake it into the validation script. GROD folks had it easy by just referencing an extent and grabbing the first cloud-free image, we have to reference the extent and the date.... I want it to be user-friendly and baking-it-in is not the best for UX.

Asks of you:
1) can you make sure you have access to all the aoi tiles? I think I made all of them public, but I'd appreciate your help making sure that's the case. 
2) could you view the sets in 4-7 and 8-9 from the outside and help me narrow down to 10 tiles/dates for the validation test? In an ideal world, these span from super easy, to kind of tricky, but don't necessarily need to cover all tiles. It would be great if there was at least one tile/date for each Landsat mission included.

Later commits Closes #16

lindsayplatt commented:
> can you make sure you have access to all the aoi tiles?

 Yes, I believe I do! I verified this by running both the `L457` and `L89` scripts in the GEE code editor, except that I commented out the `processMos()` calls because I was afraid of overwriting something accidentally.

> could you view the sets in 4-7 and 8-9 from the outside and help me narrow down to 10 tiles/dates for the validation test?

I am going through these now, but it is slow. Two questions: 
1. There's not a more clever way to do this than just editing the `var aoi = aoi2` and `var today = ee.Date(date4)` lines, right? 
2. Should I start from your list of date + aoi and try to narrow down?

steeleb commented:
Good morning!!

No worries at all on the slow, that's also where I was getting hung up. There *is* a more clever way, but it's a bit more code that I haven't added yet (this method uses the ui.Map and a dropdown - like in the maps I showed you while you were here - which is how it will be in the validation code). So I think the 'easy' way is to just edit those two variables, even though it isn't super user friendly. Since it's hard to compare in the UI, maybe just do screenshots of each of them so it's easier to compare between each tile-image? Also, the mission isn't easy to get at - it's in the first image collection that pops up in the console. 

By the way the export did work, but some of the labels are wonky. I'm going to leave it as it is for the moment, just so we can get the validation code running and then come back and fix it. It took 4-5h for each tile-image, but part of that was a silly mistake about the 'region' - and it processed the whole globe. I'll also fix that in the next round.

And don't worry about the processMos() function, you have to manually start those in the UI - and commenting works too! :)

lindsayplatt commented:
Alrighty,  I captured screenshots and info [here](https://docs.google.com/presentation/d/1ZFzmE9ap1cV1HCD4GvWwcqH5m3SavQVL_dvhs3QgmfI/edit?usp=sharing) but below are the conclusions I came to for the 10 examples. I am also wondering if we were going to share images where the clouds were already masked or if our plan was for the users to help label clouds themselves?

Easy:

- L89, Date 0, AOI 6 (open water)
- L89, Date 1, AOI 1 (easy-to-see sediment plume)
- L457, Date 0, AOI 1 (easy-to-see sediment plume, can explain sediment showing up as variety of colors)
- L457, Date 2, AOI 12 (easy-to-see sediment vs clouds)
- L457, Date 5, AOI 5 (easy-to-see sediment plume)

Intermediate:

- L89, Date 4, AOI 13 (have to zoom in to the shore)
- L89, Date 3, AOI 16 (open water, not much to reference)
- L457, Date 3, AOI 10 (“galaxy” open water/clouds mix + have to zoom to see any sediment)

Difficult:

- L457, Date 4, AOI 2 (difficult mix of clouds, sediment, land)
- L457, Date 7, AOI 4 (striping)

steeleb commented:
YOU'RE AMAZING. 

I ended up going back to the export script sooner than expected, as I needed some info from the images for the validation script and fixed the extent issue (and added the info I needed). Export is less than 1 minute now!! WAHOO!

Great question about the clouds. It seems 'easier' to me to see the cloud artifacts (dark spots) if the clouds are masked, but I agree, it might be good to provide the non-cloud-masked image as opposed to the masked image. Since we aren't particularly interested in labelling cloud artifacts (they'll likely just be considered openWater), it seems totally reasonable to leave the clouds in. I'll update on my end and use this subset you provided for the validation script. (I'm going to keep the whole list, just in case.)

steeleb commented:
Okay, latest commit plus your stellar tile-date work Closes #16  - note we still export more than the 10 tile-dates, and that's okay, I'll make the 'official' collection in the validation script.

I think this is ready to close, if you think so too! Working out the kinks of the validation script now -- the trick here is that we have to add properties for each pin of date/mission info which was not needed in GROD. I'm close to having a script that you can play with now that I have all the tile-date properties I need!

lindsayplatt reviewed (approved):
Looks good! Just wanted to point out that the `validationSet.js` file was deleted. I couldn't find a commit that deleted it explicitly so just wanted to call it out.

steeleb commented:
It was a rename! :) that spun off into `validationSet_L457.js` and `validationSet_L89.js`
